Compiler/BackEnd/Differentiate.mo

```diff
@@ -1462,7 +1462,10 @@ algorithm
       exp_1 = Expression.makeZeroExpression(Expression.arrayDimension(tp));
     then (exp_1, inFuncs);
 
-    case ("$getPart",_) then (exp, inFuncs);
+    case ("$getPart",_) equation
+        (exp_1, funcs) = differentiateExp(exp, inDiffwrtCref, inInputData,inDiffType,inFuncs, maxIter);
+    then (exp_1, inFuncs);
```

Comment:
What is `$getPart` and why aren't you returning the updated function tree?

Reply:
Oh, good point. Of course it has to be the new function tree.

As far as I understood @rfranke `$getPart` is a transition between a continuous variable and the discrete variable an an internal OpenModelica representation. The source of it, is one of the commit inside this squash: [OMCompiler/db90b5d5](https://github.com/OpenModelica/OMCompiler/commit/db90b5d56615e9487df1890824b40c9160576ff7)